tests/components/maxcube/conftest.py

```diff
@@ -0,0 +1,132 @@
+    t = create_autospec(MaxThermostat)
+    t.name = "TestThermostat"
+    t.serial = "AABBCCDD01"
+    t.rf_address = "abc1"
+    t.room_id = 1
+    t.is_thermostat.return_value = True
+    t.is_wallthermostat.return_value = False
+    t.is_windowshutter.return_value = False
+    t.mode = MAX_DEVICE_MODE_AUTOMATIC
+    t.comfort_temperature = 19.0
+    t.eco_temperature = 14.0
+    t.target_temperature = 20.5
+    t.actual_temperature = 19.0
+    t.max_temperature = None
+    t.min_temperature = None
+    t.valve_position = 25  # 25%
+    return t
+
+
+@pytest.fixture
+def wallthermostat():
+    """Create test MAX! wall thermostat."""
+    t = create_autospec(MaxWallThermostat)
+    t.name = "TestWallThermostat"
+    t.serial = "AABBCCDD02"
+    t.rf_address = "abc2"
+    t.room_id = 1
+    t.is_thermostat.return_value = False
+    t.is_wallthermostat.return_value = True
+    t.is_windowshutter.return_value = False
+    t.mode = MAX_DEVICE_MODE_MANUAL
+    t.comfort_temperature = 19.0
+    t.eco_temperature = 14.0
+    t.target_temperature = 4.5
+    t.actual_temperature = 19.0
+    t.max_temperature = 29.0
+    t.min_temperature = 4.5
+    return t
+
+
+@pytest.fixture
+def windowshutter():
+    """Create test MAX! window shutter."""
+    shutter = create_autospec(MaxWindowShutter)
+    shutter.name = "TestShutter"
+    shutter.serial = "AABBCCDD03"
+    shutter.rf_address = "abc3"
+    shutter.room_id = 1
+    shutter.is_open = True
+    shutter.is_thermostat.return_value = False
+    shutter.is_wallthermostat.return_value = False
+    shutter.is_windowshutter.return_value = True
+    return shutter
+
+
+@pytest.fixture
+def hass_config():
+    """Return test HASS configuration."""
+    return {
+        DOMAIN: {
+            "gateways": [
+                {
+                    "host": "1.2.3.4",
+                }
+            ]
+        }
+    }
+
+
+@pytest.fixture
+def hass():
+    """Mock hass core."""
+    hass = get_test_home_assistant()
+    yield hass
+    hass.stop()
+
+
+@pytest.fixture
+def cube(hass, hass_config, room, thermostat, wallthermostat, windowshutter):
+    """Build and setup a cube mock with a single room and some devices."""
+    with patch("homeassistant.components.maxcube.MaxCube") as mock:
+        cube = mock.return_value
+        cube.rooms = [room]
+        cube.devices = [thermostat, wallthermostat, windowshutter]
+        cube.room_by_id.return_value = room
+        cube.devices_by_room.return_value = [thermostat, wallthermostat, windowshutter]
+        setup_component(hass, HA_DOMAIN, {})
+        setup_component(hass, DOMAIN, hass_config)
+        hass.block_till_done()
+        gateway = hass_config[DOMAIN]["gateways"][0]
+        mock.assert_called_with(gateway["host"], gateway.get("port", 62910))
+        return cube
+
+
+def updated_state(hass, entity_id):
+    """Call update entity service and return the updated state of the given entity."""
+    hass.services.call(
```

Comment:
Move time forward instead and let the default polling update the entity.
https://github.com/home-assistant/core/blob/272dffc3843fbca32fbb4f92639c0a815753a888/tests/common.py#L363-L365

Reply:
I did not know about this feature. Using it!

Reply:
I have replaced it by:
```
        async_fire_time_changed(hass, datetime.utcnow() + timedelta(minutes=10))
        await hass.async_block_till_done()
        state = hass.states.get(ENTITY_ID)
```

But state does not get updated like it does when calling that method. What else should I do?

Reply:
Ohh, I need to use now() instead of utcnow(). Done!

Reply:
Please use our dt util as much as possible. There's a `utcnow` function there.